app/models/talk.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class Talk < ApplicationRecord
+  belongs_to :user
+
+  validates :event_name, :talk_title, :date, presence: true
+  validate :future_date?
+
+  private
+
+  def future_date?
+    return if date.blank?
+
+    errors.add(:date, I18n.t('activerecord.errors.models.talk.attributes.date.future_date')) if date.future?
```

Comment:
use a type instead adding the call to i18n here, Active record already do this https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-add

Reply:
on it